sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -69,7 +73,7 @@ def __init__(
         self,
         parsed_url,  # type: Any
         service,  # type: str
-        credential=None,  # type: Optional[Any]
+        credential=None,  # type: Optional[Union[AzureNamedKeyCredential, AzureSasCredential, Dict[str,str], str, "TokenCredential"]]
```

Comment:
Do we want to go with this ordering before I duplicate this effort (I made it alphabetical, but still not sure if we are in love with that or want to group the more obvious types at the front and then the `Credential` types at the back?

Reply:
Let's go with `str` and `Dict` first since those are probably what most people will want to use (and use today).
`str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, "TokenCredential"`?

Reply:
Will do!